### DISCUSSION TITLE
valetudo clean then go to preset

### DISCUSSION THREAD

**Discussion by cranky-coder** [Hypfer/Valetudo#1206] [Q&A / Support] [answered]
Is it possible to do a cleansegment or full clean and when done have the vacuum go to a preset (in this case, the trashcan)

I am looking to do an automation where i can tell it to clean, then go sit by the trashcan to be emptied.  

Thanks!! Great work, so far I am really enjoying it!

**Comment by ingoschi** [comment, marked as answer]
To automate this, you need an external script (or Home Assistant, openHAB, etc) that would call the GoToLocationCapability (via MQTT or REST) when the status changes from "cleaning" to "returning" (listen to MQTT /StatusStateAttribute/status or polling the REST interface).
So the robot would go to the new location instead of returning to the docking station when finished cleaning.

**Reply by cranky-coder** [reply]
I will give that a shot.  I have the vac setup in HA currently.  I didn't know if there was something specific i was missing or needed to disable return to home after zone cleaning.  Thanks!!!
